examples/Example1.elm

```diff
@@ -82,10 +82,10 @@ view model =
 dropdownConfig : Dropdown.Config String Msg
 dropdownConfig =
     let
-        itemToPrompt item =
-            Element.text item
+        itemsToPrompt items =
+            Element.text <| Maybe.withDefault "nothing selected" <| List.head items
 
         itemToElement selected highlighted item =
             Element.text item
     in
-    Dropdown.basic DropdownMsg OptionPicked itemToPrompt itemToElement
+    Dropdown.basic DropdownMsg OptionPicked itemsToPrompt itemToElement
```

Comment:
I was thinking about that solution and I find out that we will have a piece of code that almost all the tame it will be replicated by the dev user (if they are using the "single item" version):


```elm
List.head items
    |> Maybe.withDefault "my default text"
    |> Element.text
```

So I was thinking in a way to the package absorbs this and I came up with an idea but I'm not sure if actually makes sense, I will try to explain my thought.

So before this or we had 3 starting points to create a dropdown right?

- basic
- custom
- filterable

And all of them only takes into account single item selection. Switching `itemToPrompt` to `itemsToPrompt` also switches the basic assumption of being a single item dropdown to be a multi-select dropdown that also supports single item dropdown with some tweaks. So it means that we are basically treating the multi-select dropdown as a default use case.

But then, if we check the `OptionPicked`, it's only returning a Maybe item, it means that if you want to use it for a multiselect dropdown purpose you still will need to do some tweaks in the update. 

Then, because of that I was actually thinking about creating more "starting points" functions. something like:

- single
- multiselect
- custom
- singleFilterable
- multiSelectFilterable

this way, for single item dropdowns, we could maintain  the function signature:

```elm
single :
    (Msg item -> msg)
    -> (Maybe item -> msg)
    -> (List item -> Element msg)
    -> (Bool -> Bool -> item -> Element msg)
    -> Config item msg
```

and doing like what we have right now, but for multiselect we could do something like: 

```elm
single :
    (Msg item -> msg)
    -> (List item -> msg)
    -> (List item -> Element msg)
    -> (Bool -> Bool -> item -> Element msg)
    -> Config item msg
```

Notice the difference in the second argument, this way, in the update, we can return a list of the selected items. the usage could be something like:

```elm
    Dropdown.multiselect DropdownMsg OnSelectedItemsChanged itemsToPrompt itemsToElement
``` 

and then we can handle all that logic about what type of message we should trigger internally, and without having to create another module or something like that and last, but not least, it will be totally "transparent" from the user perspective.

I would like to know your thoughts about that :D not sure if those API I'm proposing actually make sense or not :D 

I noticed one day that you are in the elm slack, so you can ping me there (I think @PedroHLC  is also there)

Reply:
I see your point here and think that you are right, however, the issue is the internal representation of the State... it needs to be a list to support both cases (we are changing `Maybe` for `List` here), and this means we cannot create a config object that tales `itemToPrompt : Maybe item -> msg` and also an additional `itemsToPrompt: List item -> msg` 😟  

We can define one in terms of the other (since the `List` is more flexible) but I cannot think of a way to preserve the previous function having changed the internal representation of the `selectedItems` (I'm open for ideas here, maybe I'm too dense this morning 😅).